packages/react-ui/src/widgets/BarWidgetUI.js

```diff
@@ -0,0 +1,418 @@
+          return {
+            value,
+            ...(isDisabled && {
+              itemStyle: { color: theme.palette.charts.disabled },
+              disabled: true
+            })
+          };
+        }),
+        ...(stacked && { stack: 'total' }),
+        ...(!IS_TOUCH_SCREEN && {
+          emphasis: {
+            focus: 'series',
+            itemStyle: {
+              color: darken(colors[componentIdx] || '#000', 0.25)
+            }
+          }
+        })
+      })),
+    [animation, colors, series, yAxisData, selectedBars, stacked, theme]
+  );
+
+  const options = useMemo(
+    () => ({
+      grid: {
+        left: theme.spacing(0.5),
+        top: theme.spacing(2),
+        right: vertical ? theme.spacing(0.5) : theme.spacing(5),
+        bottom: theme.spacing(0),
+        containLabel: true
+      },
+      axisPointer: {
+        lineStyle: {
+          color: theme.palette.charts.axisPointer
+        }
+      },
+      color: colors,
+      tooltip: tooltipOptions,
+      xAxis: vertical ? xAxisOptions : yAxisOptions,
+      yAxis: vertical ? yAxisOptions : xAxisOptions,
+      series: seriesOptions
+    }),
+    [theme, colors, tooltipOptions, xAxisOptions, yAxisOptions, seriesOptions, vertical]
+  );
+
+  const clearBars = () => {
+    if (onSelectedBarsChange) {
+      onSelectedBarsChange([], []);
+    }
+  };
+
+  const clickEvent = useCallback(
+    (params) => {
+      if (onSelectedBarsChange) {
+        const selectedBarsCp = [...selectedBars];
+        const { dataIndex, componentIndex } = params;
+
+        const selectedIdx = selectedBarsCp.findIndex(
+          ([sDataIndex, sComponentIndex = 0]) =>
+            dataIndex === sDataIndex && componentIndex === sComponentIndex
+        );
+
+        if (selectedIdx === -1) {
+          selectedBarsCp.push([dataIndex, componentIndex]);
+        } else {
+          selectedBarsCp.splice(selectedIdx, 1);
+        }
+
+        const selectedBarsAsObject = selectedBarsCp.map(
+          ([sDataIndex, sComponentIndex = 0]) => ({
+            xAxis: vertical ? xAxisData[sDataIndex] : series[sDataIndex],
+            yAxis: vertical ? series[sComponentIndex] : xAxisData[sComponentIndex],
+            value: yAxisData[sComponentIndex][sDataIndex]
+          })
+        );
+        onSelectedBarsChange(selectedBarsCp, selectedBarsAsObject);
+      }
+    },
+    [yAxisData, onSelectedBarsChange, selectedBars, vertical, xAxisData, series]
+  );
+
+  const onEvents = useMemo(
+    () => ({
+      click: clickEvent
+    }),
+    [clickEvent]
+  );
+
+  return (
+    <div>
+      {onSelectedBarsChange && (
+        <Grid
+          container
+          direction='row'
+          justifyContent='space-between'
+          alignItems='center'
+          className={classes.optionsSelectedBar}
+        >
+          <Typography variant='caption'>
+            {selectedBars && selectedBars.length ? selectedBars.length : 'All'} selected
```

Comment:
`selectedBars` defaults to `[]`, so you can simplify this expression to just check the length (also below).
Or we want extra-safety, the `?` operator.

Reply:
It defaults to [] when it's undefined, but not when it's null if I don't remember bad.

Reply:
Indeed. So `?` will solve that too :)

Reply:
Agree 👌